x/superfluid/keeper/concentrated_liquidity.go

```diff
@@ -46,6 +48,11 @@ func (k Keeper) addToConcentratedLiquiditySuperfluidPosition(ctx sdk.Context, ow
 		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, 0, types.PositionNotSuperfluidStakedError{PositionId: position.PositionId}
 	}
 
+	// Defense in depth making sure that the position is full-range.
+	if position.LowerTick != cltypes.MinTick || position.UpperTick != cltypes.MaxTick {
+		return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, 0, types.ConcentratedTickRangeNotFullError{ActualLowerTick: position.LowerTick, ActualUpperTick: position.UpperTick}
+	}
+
```

Comment:
nit: maybe we can extract this validation with the validation below this function into a helper function to make this easier to parse.